agent/osv_agent.py

```diff
@@ -20,6 +43,62 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_content(message: m.Message) -> bytes | None:
+    """Get the content of the file from the message.
+    Args:
+        message: The message containing the file content.
+    Returns:
+        The content of the file.
+    """
+    content = message.data.get("content")
+    if content is not None:
+        casted_content = typing.cast(bytes, content)
+        return casted_content
+    content_url = message.data.get("content_url")
+    if content_url is not None:
+        return requests.get(content_url, timeout=60).content
+    return None
+
+
+def _construct_commands(file_path: str) -> list[list[str]]:
+    """Constructs OSV command, for both lockfile and sbomfile."""
+    return [
+        [
+            "/usr/local/bin/osv-scanner",
+            "--format",
+            "json",
+            "--lockfile",
+            file_path,
+        ],
+        [
+            "/usr/local/bin/osv-scanner",
+            "--format",
+            "json",
+            "--sbom",
+            file_path,
+        ],
+    ]
+
+
+def _run_osv(path: str, content: bytes) -> str | None:
+    """Perform the osv scan with two flags with --sbom and --lockfile,  letting OSV validate the file
+     instead of guessing the file format.
+    Args:
+        path: The file path.
+        content: Scanned file content
+    """
+    file_name = pathlib.Path(path).name
+    decoded_content = content.decode("utf-8")
+    with open(file_name, "w", encoding="utf-8") as file_path:
+        file_path.write(decoded_content)
+
+    for command in _construct_commands(file_name):
+        output = _run_command(command)
+        if output is not None:
+            return output
```

Comment:
Does it make sense to bailout of the first one to not return None?

Reply:
The output gets validated before returning None.

Reply:
ok.